python/ray/train/callbacks/results_preprocessors/average.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from typing import Dict, List, Tuple
+
+import numpy as np
+
+from ray.train.callbacks.results_preprocessors.preprocessor import ResultsPreprocessor
+
+
+logger = logging.getLogger(__name__)
+
+
+class AverageResultsPreprocessor(ResultsPreprocessor):
+    """A preprocessor that average training metrics from all workers.
+
+    Args:
+        metrics_to_average (Dict): A Dict of metrics in results to average.
+            The key is the metric to be averaged across all workers. The value
+            is the magic key in the results that will be used as weights. If
+            the value is None, the weight will be taken to be equal for all
+            workers. Both key and value should be reported using `train.report()`.
+    """
+
+    VALID_SUMMARY_TYPES: Tuple[type] = (
+        int,
+        float,
+        np.float32,
+        np.float64,
+        np.int32,
+        np.int64,
+    )
+
+    def __init__(self, metrics_to_average: Dict = None):
+        self.metrics_to_average = (
+            {} if metrics_to_average is None else metrics_to_average
```

Comment:
@matthewdeng Thank you for your review! I didn't include a default behavior here, because it is not clear to me what default metrics are good for averaging (iteration time is a good one). Looking for some suggestions and comments from @Yard1 @amogkam.